azurerm_container_group will panic with *terraform.EvalSequence, err: rpc error: code = Unavailable desc = transport is closing

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform (and AzureRM Provider) Version

1.39

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* `azurerm_container_group`

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
resource "azurerm_container_group" "master" {
  name                = "epijmeter-master"
  location            = "${azurerm_resource_group.jmeter_rg.location}"
  resource_group_name = "${azurerm_resource_group.jmeter_rg.name}"

  ip_address_type = "private"
  os_type         = "Linux"

  network_profile_id = "${azurerm_network_profile.example.id}"

  restart_policy = "Never"

  container {
    name   = "jmeter"
    image  = "${var.jmeter_docker_image}"
    cpu    = "4.0"
    memory = "16.0"

    ports {
      port     = 1099
      protocol = "TCP"
    }

    volume {
      name                 = "jmeter"
      mount_path           = "/jmeter"
      read_only            = false
      storage_account_name = "${azurerm_storage_account.example.name}"
      storage_account_key  = "${azurerm_storage_account.example.primary_access_key}"
      share_name           = "${azurerm_storage_share.example.name}"
    }

    commands = [
      "/bin/sh",
      "-c",
      "cd /jmeter; jmeter -n -Jserver.rmi.ssl.disable=true -t DocumentApiMod_addNew.jmx -l test.jtl -e -o report -R ${join(",", "${azurerm_container_group.slaves.*.ip_address}")}",
    ]
  }
  depends_on = [azurerm_storage_account.example, azurerm_storage_share.example]
}

```

### Debug Output

```
2020/01/03 21:30:45 [ERROR] <root>: eval: *terraform.EvalRefresh, err: rpc error: code = Unavailable desc = transport is closing 
2020/01/03 21:30:45 [ERROR] <root>: eval: *terraform.EvalSequence, err: rpc error: code = Unavailable desc = transport is closing 
2020/01/03 21:30:45 [TRACE] [walkRefresh] Exiting eval tree: azurerm_container_group.master 
2020/01/03 21:30:45 [TRACE] vertex "azurerm_container_group.master": visit complete 
2020/01/03 21:30:45 [TRACE] vertex "azurerm_container_group.master": dynamic subgraph encountered errors 
```

### Panic Output

[crash.log](https://github.com/terraform-providers/terraform-provider-azurerm/files/4021161/crash.log)

### Expected Behavior

execute terraform apply twice, it works fine.

### Actual Behavior

execute terraform apply twice, it will panic. 

### Steps to Reproduce


1. terraform apply -auto-approve -var 'client_id=$(ARM_CLIENT_ID)' -var 'subscription_id=$(ARM_SUBSCRIPTION_ID)' -var 'client_secret=$(ARM_CLIENT_SECRET)' -var 'tenant_id=$(ARM_TENANT_ID)' -var 'jmeter_docker_image=$(image_name)'

2. execute the same terraform apply as 1. -> panic

### Important Factoids

no

### References
The root cause might be this line. According to the log, the response doesn't include IPAddress on property.

https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/internal/services/containers/resource_arm_container_group.go#L581

* #0000

--- a/azurerm/internal/services/containers/resource_arm_container_group.go
+++ b/azurerm/internal/services/containers/resource_arm_container_group.go
@@ -578,7 +578,7 @@ func resourceArmContainerGroupRead(d *schema.ResourceData, meta interface{}) err
 	}
 
 	if props := resp.ContainerGroupProperties; props != nil {
-		containerConfigs := flattenContainerGroupContainers(d, resp.Containers, props.IPAddress.Ports, props.Volumes)
+		containerConfigs := flattenContainerGroupContainers(d, resp.Containers, props.IPAddress, props.Volumes)
 		if err := d.Set("container", containerConfigs); err != nil {
 			return fmt.Errorf("Error setting `container`: %+v", err)
 		}
@@ -1104,7 +1104,7 @@ func flattenContainerImageRegistryCredentials(d *schema.ResourceData, input *[]c
 	return output
 }
 
-func flattenContainerGroupContainers(d *schema.ResourceData, containers *[]containerinstance.Container, containerGroupPorts *[]containerinstance.Port, containerGroupVolumes *[]containerinstance.Volume) []interface{} {
+func flattenContainerGroupContainers(d *schema.ResourceData, containers *[]containerinstance.Container, ipAddress *containerinstance.IPAddress, containerGroupVolumes *[]containerinstance.Volume) []interface{} {
 	//map old container names to index so we can look up things up
 	nameIndexMap := map[string]int{}
 	for i, c := range d.Get("container").([]interface{}) {
@@ -1166,13 +1166,16 @@ func flattenContainerGroupContainers(d *schema.ResourceData, containers *[]conta
 			containerConfig["port"] = containerPort
 			// protocol isn't returned in container config, have to search in container group ports
 			protocol := ""
-			if containerGroupPorts != nil {
-				for _, cgPort := range *containerGroupPorts {
-					if *cgPort.Port == containerPort {
-						protocol = string(cgPort.Protocol)
+			if ipAddress != nil {
+				if containerGroupPorts := ipAddress.Ports; containerGroupPorts != nil {
+					for _, cgPort := range *containerGroupPorts {
+						if *cgPort.Port == containerPort {
+							protocol = string(cgPort.Protocol)
+						}
 					}
 				}
 			}
+
 			if protocol != "" {
 				containerConfig["protocol"] = protocol
 			}

--- a/azurerm/internal/services/containers/resource_arm_container_group.go
+++ b/azurerm/internal/services/containers/resource_arm_container_group.go
@@ -1175,7 +1175,6 @@ func flattenContainerGroupContainers(d *schema.ResourceData, containers *[]conta
 					}
 				}
 			}
-
 			if protocol != "" {
 				containerConfig["protocol"] = protocol
 			}

